### PR TITLE
Refactor `reserve-excess-capacity` Helm chart into Golang component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test enhancement
/topology seed
/platform all
/exp intermediate
/merge squash

**What this PR does / why we need it**:
This PR refactors the deployment for the `reserve-excess-capacity` Helm resources. It adopts the component concept. Consequently, it eliminates the Helm resources and adds unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754 

**Release note**:

```improvement user
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/component/seedsystem/seedsystem.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ManagedResourceName is the name of the ManagedResource containing the resource specifications.
+const ManagedResourceName = "system"
+
+// Values is a set of configuration values for the system resources.
+type Values struct {
+	// ReserveExcessCapacity contains configuration for the deployment of the excess capacity reservation resources.
+	ReserveExcessCapacity ReserveExcessCapacityValues
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/seedsystem/seedsystem.go:41]: Can you elaborate why you picked the approach with "seedsystem" component over the one having a separate component for reserve-excess-capacity? Do we expect new "components" to be added to this "seedsystem" component?

File: pkg/operation/botanist/component/seedsystem/seedsystem.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+func (s *seedSystem) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, s.client, s.namespace, ManagedResourceName)
+}
+
+func (s *seedSystem) computeResourcesData() (map[string][]byte, error) {
+	var (
+		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
```
Comment (ary1992) [inline pkg/operation/botanist/component/seedsystem/seedsystem.go]: Shouldn't we use following:
```suggestion
		registry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
```

**Review by ary1992 — COMMENTED** [review]
Minor suggestion while reviewing the PR with @ialidzhikov

**Review by ialidzhikov — COMMENTED** [review]
1 inline question. 

I was also wondering what is the difference between the reserve-excess-capacity that is deployed as part of the Seed reconciliation and the following extension https://github.com/gardener/reserve-excess-capacity?

File: pkg/operation/botanist/component/seedsystem/seedsystem.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/component/seedsystem/seedsystem.go:41]: > Can you elaborate why you picked the approach with "seedsystem" component over the one having a separate component for reserve-excess-capacity? Do we expect new "components" to be added to this "seedsystem" component?

We could have also created a dedicated package just for this, but given that it's only two resources I thought about having something similar to https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/component/shootsystem will make it easier to maintain such "small things" for us during development. For example, the new `PriorityClasses` created as part of #5652 can be added here as well.
Larger components should still get their own package to have a clear distinction.

> I was also wondering what is the difference between the reserve-excess-capacity that is deployed as part of the Seed reconciliation and the following extension https://github.com/gardener/reserve-excess-capacity?

I think https://github.com/gardener/reserve-excess-capacity can be used to also deploy the capacity reservation pods on dedicated worker pools for the seeds.

**Review by ialidzhikov — COMMENTED** [review]
/lgtm

**Pull request merged**